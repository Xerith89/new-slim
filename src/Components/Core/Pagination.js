import React, { Component } from 'react'
import TableHeader from './TableHeader'
import Spinner from './Spinner'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faChevronLeft, faChevronRight} from '@fortawesome/free-solid-svg-icons'

export default class Pagination extends Component {
    constructor(props) {
        super(props)
        this.state = {
            currentPage: 1,
            totalRecords: this.props.filteredData.length,
            paginatedData: this.props.filteredData.slice(this.currentPage-1,this.props.recordsPerPage),
            finalPage: Math.ceil(this.props.filteredData.length/this.props.recordsPerPage)
        }
    }

    componentDidUpdate(prevProps) {
        if(prevProps.filteredData !== this.props.filteredData) {
          this.setState({paginatedData: this.props.filteredData});
        }
      }

    handleClick = (event) => {
       
        if (event.currentTarget.name === 'next') {
            this.setState({
                currentPage: this.state.currentPage+1,
                paginatedData: this.props.filteredData.slice(this.state.currentPage*this.props.recordsPerPage,(this.state.currentPage*this.props.recordsPerPage)+this.props.recordsPerPage),
            })
        } else if (event.currentTarget.name === 'previous') {
           
            this.setState({
                currentPage: this.state.currentPage - 1,
                paginatedData: this.props.filteredData.slice(this.state.currentPage/this.props.recordsPerPage,(this.state.currentPage/this.props.recordsPerPage)+this.props.recordsPerPage),
            })
        }
        else {
            let page = parseInt(event.target.value)*this.props.recordsPerPage;
            console.log(page);
        }
    }


    render() {
        let pageNumbers = 1;
        return (
            <React.Fragment>
                <table className="table table-hover table-bordered">
                    <TableHeader fetching={this.props.fetching} tableHeaderOptions={this.props.tableHeaderOptions} paginatedData={this.state.paginatedData}/> 
                </table>
                {this.props.fetching ? <Spinner /> : null}          
                <div> 
                    <nav aria-label="Page navigation example">
                        <ul className="pagination justify-content-center">
                            <li className="page-item">
                            {this.state.currentPage === 1 ? <button name="previous" onClick={this.handleClick} className="btn disabled" aria-label="Previous" disabled><span aria-hidden="true"><FontAwesomeIcon icon={faChevronLeft} /></span></button> :
                            <button name="previous" onClick={this.handleClick} className="page-link" href="/" aria-label="Previous"><span aria-hidden="true"><FontAwesomeIcon icon={faChevronLeft} /></span></button>} 
                            </li>
                            {this.state.finalPage > 1 ? <li className="page-item"><button className="page-link" value={pageNumbers} onClick={this.handleClick}>{pageNumbers}</button></li> : null} 
                            <li className="page-item">
                            {this.state.currentPage !== this.state.finalPage ? <button name="next" onClick={this.handleClick}className="page-link" aria-label="Next">
                                <span aria-hidden="true"><FontAwesomeIcon icon={faChevronRight} /></span>
                            </button> : <button name="next" onClick={this.handleClick}className="btn disabled" aria-label="Next" disabled>
                                <span aria-hidden="true"><FontAwesomeIcon icon={faChevronRight} /></span>
                            </button>}
                            </li>
                        </ul>
                    </nav>
                </div>
            </React.Fragment>
        )
    }
}

Pagination.propTypes = {
    fetchingTasks: PropTypes.bool,
    filteredData: PropTypes.array,
    totalRecords: PropTypes.number,
    recordsPerPage: PropTypes.number
}