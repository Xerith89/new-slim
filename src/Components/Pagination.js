import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faChevronLeft, faChevronRight} from '@fortawesome/free-solid-svg-icons'

export default class Pagination extends Component {
    constructor(props) {
        super(props)
        this.state = {
            currentPage: 1,
            totalRecords: this.props.totalRecords,
            paginatedData: [],
            lastPage: 0
        }
      
    }

    componentDidUpdate(prevProps) {
        if(prevProps.data !== this.props.data) {
          this.setState({
            paginatedData: this.props.data.slice(this.currentPage-1,this.props.recordsPerPage), 
            totalRecords:this.props.totalRecords,
            currentPage: 1,
            lastPage: Math.ceil(this.props.data.length/this.props.recordsPerPage)
        });
        }
    }

    calculateRange = () => {
        //final page +1 will mean our for loop stops are the final page
        return Math.min((this.props.range + this.state.currentPage), this.state.lastPage+1);
    }


    handleClick = (event) => {

        const {recordsPerPage, data} = this.props;
        switch (event.currentTarget.name) {
        case 'nextPage': 
            this.setState({
                currentPage: this.state.currentPage === this.state.lastPage ? this.state.currentPage : this.state.currentPage+1,
                paginatedData: data.slice(this.state.currentPage*recordsPerPage,(this.state.currentPage*recordsPerPage)+recordsPerPage),
            });
            break;
        case 'firstPage' :
            this.setState({
                currentPage: 1,
                paginatedData: data.slice(0,0+recordsPerPage),
            });
            break;
        case 'previousPage':
            const offset = (this.state.currentPage-2)*recordsPerPage
            this.setState({
                currentPage: this.state.currentPage === 1 ? this.state.currentPage : this.state.currentPage-1,
                paginatedData: data.slice(offset,offset+recordsPerPage),
            });
            break;
        case 'lastPage':
            this.setState({
                currentPage: this.state.lastPage,
                paginatedData: data.slice(this.state.lastPage*recordsPerPage-recordsPerPage, this.state.lastPage*recordsPerPage+recordsPerPage),
            });
            break;
       default:
            let page;
            try {
            page = parseInt(event.target.value);
            } catch (error) { console.log(error);}
            const pageOffset = (page*recordsPerPage)-recordsPerPage;
            this.setState({
                currentPage: page,
                paginatedData: data.slice(pageOffset,pageOffset+recordsPerPage),
            });
            break;
        }
    }

    render() {
        //To be extracted into something a bit cleaner and more functional
        let pages = [];
        if (this.props.range !== null) {
            //Subtracting range ensures there are always x many pages displayed where x refers to the range
            for(let i = (this.calculateRange()-this.props.range); i < this.calculateRange(); i++) {
                if (i <= this.state.lastPage) {
                    pages.push(i);
                }
            }
        } else {
            for(let i = 1; i <= this.state.lastPage; i++) {
                pages.push(i);
            }
        }
        //

        let newChild = React.cloneElement(this.props.children, {
            paginatedData: this.state.paginatedData});

        let pagebutton;
        return (
            <div> 
                <div>{newChild}</div>
                <nav aria-label="Page navigation example">
                    <ul className="pagination justify-content-center">

                        <li className="page-item">
                        {this.state.currentPage === 1 ?
                        <button id="firstPage" name="firstPage" onClick={this.handleClick} className="disabled" style={{border: '0', background: 'none'}} aria-label="First" disabled><span aria-hidden="true"><FontAwesomeIcon icon={faChevronLeft} /><FontAwesomeIcon icon={faChevronLeft} /></span></button> :
                        <button name="firstPage" onClick={this.handleClick} style={{border: '0', background: 'none'}} className="page-link" href="/" aria-label="First"><span aria-hidden="true"><FontAwesomeIcon icon={faChevronLeft} /><FontAwesomeIcon icon={faChevronLeft} /></span></button>
                        }</li>
                      
                        <li className="page-item">
                        {this.state.currentPage === 1 ? <button id="previousPage" name="previousPage" onClick={this.handleClick} className="disabled" style={{border: '0', background: 'none'}} aria-label="Previous" disabled><span aria-hidden="true"><FontAwesomeIcon icon={faChevronLeft} /></span></button> :
                        <button name="previousPage" onClick={this.handleClick} style={{border: '0', background: 'none'}} className="page-link" href="/" aria-label="Previous"><span aria-hidden="true"><FontAwesomeIcon icon={faChevronLeft} /></span></button>} 
                        </li>
                        
                        <li className="page-item">
                        {this.props.range !== null && this.state.currentPage !== 1 ? "..." : null}
                        </li>
            
                        {pages.map((value) => {
                        value !== this.state.currentPage ? pagebutton = <li key={value} className="page-item"><button name={`value${value}`} style={{border: '0', background: 'none'}} className="page-link" value={value} onClick={this.handleClick}>{value}
                        </button></li> : pagebutton = <li key={value} className="page-item"><button name={`value${value}`} className="btn" value={value} style={{border: '0', background: 'none'}} onClick={this.handleClick} disabled><strong>{value}</strong></button></li>
                            return (pagebutton)
                        })}

                        <li className="page-item">
                        {this.props.range !== null && pages.includes(this.state.lastPage) ? null : "..."}
                        </li>
                        
                        {this.state.currentPage < this.state.lastPage ? <button id="nextPage"style={{border: '0',  background: 'none'}}  name="nextPage" onClick={this.handleClick}className="page-link" aria-label="Next">
                            <span aria-hidden="true"><FontAwesomeIcon icon={faChevronRight} /></span>
                        </button> : <button name="nextPage" onClick={this.handleClick}className="btn disabled" aria-label="Next" disabled>
                            <span aria-hidden="true"><FontAwesomeIcon icon={faChevronRight} /></span>
                        </button>}

                        <li className="page-item">
                        {this.state.currentPage < this.state.lastPage ?
                        <button id="lastPage" name="lastPage" onClick={this.handleClick} style={{border: '0', background: 'none'}} aria-label="Last" ><span aria-hidden="true"><FontAwesomeIcon icon={faChevronRight} /><FontAwesomeIcon icon={faChevronRight} /></span></button> :
                        <button id="lastPage" name="lastPage" onClick={this.handleClick} style={{border: '0', background: 'none'}} className="page-link disabled" href="/" aria-label="Last"><span aria-hidden="true"><FontAwesomeIcon icon={faChevronRight} /><FontAwesomeIcon icon={faChevronRight} disabled /></span></button>
                        }</li>
                        
                    </ul>
                </nav>
            </div>
        )
    }
}

Pagination.propTypes = {
    data: PropTypes.array.isRequired,
    className: PropTypes.string,
    recordsPerPage: PropTypes.number.isRequired,
    range: PropTypes.number,
    style: PropTypes.object,
    children: PropTypes.any.isRequired
}