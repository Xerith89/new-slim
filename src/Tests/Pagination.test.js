import Pagination from '../Components/Pagination'
import React from 'react'
import {shallow, configure,} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const paginatedData = [
    {
    "userId": 1,
    "id": 1,
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
    },
    {
    "userId": 1,
    "id": 2,
    "title": "qui est esse",
    "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
    },
    {
    "userId": 1,
    "id": 3,
    "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
    "body": "et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut"
    },
]

const mock = function Hello(props) {
    return (
        <div> 
            <ul>
            {props.paginatedData && props.paginatedData.map((item) => {
                return (
                    <li key={item.id}>{item.title}</li> 
                )
            })}
            </ul>
        </div>
    )
}

describe('Pagination Buttons Behave As Expected', () => {

    test('Handle click is fired on nextPage click', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3} recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.instance().handleClick = jest.fn();
        wrapper.instance().forceUpdate();
        expect(wrapper.instance().handleClick).not.toHaveBeenCalled();
        wrapper.find({name: 'nextPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalled();
        wrapper.find({name: 'nextPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalledTimes(2);
        wrapper.unmount();
    });

    test('Handle click is fired on firstPage click', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.instance().handleClick = jest.fn();
        wrapper.instance().forceUpdate();
        expect(wrapper.instance().handleClick).not.toHaveBeenCalled();
        wrapper.find({name: 'firstPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalled();
        wrapper.find({name: 'firstPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalledTimes(2);
        wrapper.unmount();
    });

    test('Handle click is fired on previousPage click', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.instance().handleClick = jest.fn();
        wrapper.instance().forceUpdate();
        expect(wrapper.instance().handleClick).not.toHaveBeenCalled();
        wrapper.find({name: 'previousPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalled();
        wrapper.find({name: 'previousPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalledTimes(2);
        wrapper.unmount();
    });

    test('Handle click is fired on lastPage click', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.setState({ lastPage: 3 });
        wrapper.instance().handleClick = jest.fn();
        wrapper.instance().forceUpdate();
        expect(wrapper.instance().handleClick).not.toHaveBeenCalled();
        wrapper.find({name: 'lastPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalled();
        wrapper.find({name: 'lastPage'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalledTimes(2);
        wrapper.unmount();
    });

    test('nextPage click increases page', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.setState({ lastPage: 3 });
        expect(wrapper.state('currentPage')).toBe(1);
        wrapper.find({name: 'nextPage'}).simulate('click', {currentTarget: { name: 'nextPage'}});
        expect(wrapper.state('currentPage')).toBe(2);
        wrapper.find({name: 'nextPage'}).simulate('click', {currentTarget: { name: 'nextPage'}});
        expect(wrapper.state('currentPage')).toBe(3);
        wrapper.find({name: 'nextPage'}).simulate('click', {currentTarget: { name: 'nextPage'}});
        expect(wrapper.state('currentPage')).toBe(3);
        wrapper.unmount();
    });

    test('previousPage click decreases page', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.setState({ currentPage: 3 });
        expect(wrapper.state('currentPage')).toBe(3);
        wrapper.find({name: 'previousPage'}).simulate('click', {currentTarget: { name: 'previousPage'}});
        expect(wrapper.state('currentPage')).toBe(2);
        wrapper.find({name: 'previousPage'}).simulate('click', {currentTarget: { name: 'previousPage'}});
        expect(wrapper.state('currentPage')).toBe(1);
        wrapper.find({name: 'previousPage'}).simulate('click', {currentTarget: { name: 'previousPage'}});
        expect(wrapper.state('currentPage')).toBe(1);
        wrapper.unmount();
    });

    test('firstPage click takes you to first page', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3} recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.setState({ currentPage: 3 });
        expect(wrapper.state('currentPage')).toBe(3);
        wrapper.find({name: 'firstPage'}).simulate('click', {currentTarget: { name: 'firstPage'}});
        expect(wrapper.state('currentPage')).toBe(1);
        wrapper.find({name: 'firstPage'}).simulate('click', {currentTarget: { name: 'firstPage'}});
        expect(wrapper.state('currentPage')).toBe(1);       
        wrapper.unmount();
    });

    test('lastPage click takes you to last page', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.setState({ lastPage: 3 });
        expect(wrapper.state('currentPage')).toBe(1);
        wrapper.find({name: 'lastPage'}).simulate('click', {currentTarget: { name: 'lastPage'}});
        expect(wrapper.state('currentPage')).toBe(3);
        wrapper.find({name: 'lastPage'}).simulate('click', {currentTarget: { name: 'lastPage'}});
        expect(wrapper.state('currentPage')).toBe(3);       
        wrapper.unmount();
    });

    test('Handleclick is fired on page number click', () => {
        let wrapper = shallow(<Pagination  data={paginatedData} totalRecords={3}  recordsPerPage={1} range={1}>{mock}</Pagination>);
        wrapper.setState({ lastPage: 3});
        wrapper.instance().handleClick = jest.fn();
        wrapper.instance().forceUpdate();
        expect(wrapper.instance().handleClick).not.toHaveBeenCalled();
        wrapper.find({name: 'value1'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalled();
        wrapper.find({name: 'value1'}).simulate('click');
        expect(wrapper.instance().handleClick).toHaveBeenCalledTimes(2);
        wrapper.unmount();
    });
});