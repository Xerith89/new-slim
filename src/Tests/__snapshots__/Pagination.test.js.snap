// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Renders to the DOM 1`] = `
Pagination {
  "context": Object {},
  "handleClick": [Function],
  "props": Object {
    "filteredData": Array [
      Object {
        "assignedTo": "Alex Coupe",
        "claimSpecNo": "5500001",
        "dueDate": "01-09-2019",
        "id": 1,
        "priority": "Normal",
        "taskName": "Review New Claim",
        "type": "Claim",
      },
      Object {
        "assignedTo": "Alex Coupe",
        "claimSpecNo": "5500000",
        "dueDate": "01-09-2019",
        "id": 2,
        "priority": "Normal",
        "taskName": "Review New Invoice",
        "type": "Claim",
      },
      Object {
        "assignedTo": "Donald Duck",
        "claimSpecNo": "5500244",
        "dueDate": "29-09-2019",
        "id": 3,
        "priority": "Urgent",
        "taskName": "Diary Review",
        "type": "Claim",
      },
      Object {
        "assignedTo": "Peter Parker",
        "claimSpecNo": "5500044",
        "dueDate": "05-09-2019",
        "id": 4,
        "priority": "Urgent",
        "taskName": "Loss Adjustor Report Received",
        "type": "Claim",
      },
    ],
    "recordsPerPage": 3,
    "totalRecords": 4,
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "currentPage": 1,
    "finalPage": 2,
    "paginatedData": Array [
      Object {
        "assignedTo": "Alex Coupe",
        "claimSpecNo": "5500001",
        "dueDate": "01-09-2019",
        "id": 1,
        "priority": "Normal",
        "taskName": "Review New Claim",
        "type": "Claim",
      },
      Object {
        "assignedTo": "Alex Coupe",
        "claimSpecNo": "5500000",
        "dueDate": "01-09-2019",
        "id": 2,
        "priority": "Normal",
        "taskName": "Review New Invoice",
        "type": "Claim",
      },
      Object {
        "assignedTo": "Donald Duck",
        "claimSpecNo": "5500244",
        "dueDate": "29-09-2019",
        "id": 3,
        "priority": "Urgent",
        "taskName": "Diary Review",
        "type": "Claim",
      },
    ],
    "totalRecords": 4,
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
      },
      "_element": <Pagination
        filteredData={
          Array [
            Object {
              "assignedTo": "Alex Coupe",
              "claimSpecNo": "5500001",
              "dueDate": "01-09-2019",
              "id": 1,
              "priority": "Normal",
              "taskName": "Review New Claim",
              "type": "Claim",
            },
            Object {
              "assignedTo": "Alex Coupe",
              "claimSpecNo": "5500000",
              "dueDate": "01-09-2019",
              "id": 2,
              "priority": "Normal",
              "taskName": "Review New Invoice",
              "type": "Claim",
            },
            Object {
              "assignedTo": "Donald Duck",
              "claimSpecNo": "5500244",
              "dueDate": "29-09-2019",
              "id": 3,
              "priority": "Urgent",
              "taskName": "Diary Review",
              "type": "Claim",
            },
            Object {
              "assignedTo": "Peter Parker",
              "claimSpecNo": "5500044",
              "dueDate": "05-09-2019",
              "id": 4,
              "priority": "Urgent",
              "taskName": "Loss Adjustor Report Received",
              "type": "Claim",
            },
          ]
        }
        recordsPerPage={3}
        totalRecords={4}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <React.Fragment>
        <table
          className="table table-hover table-bordered"
        >
          <TableHeader
            paginatedData={
              Array [
                Object {
                  "assignedTo": "Alex Coupe",
                  "claimSpecNo": "5500001",
                  "dueDate": "01-09-2019",
                  "id": 1,
                  "priority": "Normal",
                  "taskName": "Review New Claim",
                  "type": "Claim",
                },
                Object {
                  "assignedTo": "Alex Coupe",
                  "claimSpecNo": "5500000",
                  "dueDate": "01-09-2019",
                  "id": 2,
                  "priority": "Normal",
                  "taskName": "Review New Invoice",
                  "type": "Claim",
                },
                Object {
                  "assignedTo": "Donald Duck",
                  "claimSpecNo": "5500244",
                  "dueDate": "29-09-2019",
                  "id": 3,
                  "priority": "Urgent",
                  "taskName": "Diary Review",
                  "type": "Claim",
                },
              ]
            }
          />
        </table>
        <div>
          <nav
            aria-label="Page navigation example"
          >
            <ul
              className="pagination justify-content-center"
            >
              <li
                className="page-item"
              >
                <button
                  aria-label="Previous"
                  className="btn disabled"
                  disabled={true}
                  id="previousPage"
                  name="previousPage"
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className=""
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f053",
                            "M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z",
                          ],
                          "iconName": "chevron-left",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    />
                  </span>
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  className="btn disabled"
                  disabled={true}
                  onClick={[Function]}
                  value={1}
                >
                  <strong>
                    1
                  </strong>
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  className="page-link"
                  name="value2"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": "none",
                      "border": "0",
                    }
                  }
                  value={2}
                >
                  2
                </button>
              </li>
              <button
                aria-label="Next"
                className="page-link"
                id="nextPage"
                name="nextPage"
                onClick={[Function]}
                style={
                  Object {
                    "background": "none",
                    "border": "0",
                  }
                }
              >
                <span
                  aria-hidden="true"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          320,
                          512,
                          Array [],
                          "f054",
                          "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                        ],
                        "iconName": "chevron-right",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                </span>
              </button>
            </ul>
          </nav>
        </div>
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
